File: app/src/main/java/com/m7mdabaza/fooddelivery/ui/adapters/PopularFoodAdapter.java
package com.m7mdabaza.fooddelivery.ui.adapters;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


import com.m7mdabaza.fooddelivery.R;
import com.m7mdabaza.fooddelivery.ui.models.PopularModel;

import java.util.ArrayList;

public class PopularFoodAdapter extends RecyclerView.Adapter<PopularFoodAdapter.PopularFoodViewHolder> {

    private ArrayList<PopularModel> PopularModel = new ArrayList<>();


    @NonNull
    @Override
    public PopularFoodViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.popular_food_item, parent, false);
        return new PopularFoodViewHolder(view);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull PopularFoodViewHolder holder, int position) {
        PopularModel popularModel = PopularModel.get(position);
        holder.foodName.setText(popularModel.getFoodName());
        holder.foodDescription.setText(popularModel.getFoodDescription());
        holder.foodRate.setText(popularModel.getRate() + "");
    }

    @Override
    public int getItemCount() {
        return PopularModel.size();
    }

    static class PopularFoodViewHolder extends RecyclerView.ViewHolder {

        ImageView imageUrl;
        TextView foodName, foodDescription, foodRate;

        PopularFoodViewHolder(@NonNull View itemView) {
            super(itemView);

            imageUrl = itemView.findViewById(R.id.foodImage);
            foodName = itemView.findViewById(R.id.foodName_tv);
            foodDescription = itemView.findViewById(R.id.foodDesc_tv);
            foodRate = itemView.findViewById(R.id.rate_tv);

        }

    }

    public void setList(ArrayList<PopularModel> PopularModel) {
        this.PopularModel = PopularModel;

    }
}

File: app/src/main/java/com/m7mdabaza/fooddelivery/adapters/PopularFoodAdapter.kt
package com.m7mdabaza.fooddelivery.adapters

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.m7mdabaza.fooddelivery.R
import com.m7mdabaza.fooddelivery.models.PopularModel
import java.util.*

class PopularFoodAdapter : RecyclerView.Adapter<PopularFoodAdapter.PopularFoodViewHolder>() {

    private var popularModel = ArrayList<PopularModel>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PopularFoodViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.popular_food_item, parent, false)
        return PopularFoodViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: PopularFoodViewHolder, position: Int) {
        val popularModel = popularModel[position]
        holder.foodName.text = popularModel.foodName
        holder.foodDescription.text = popularModel.foodDescription
        holder.foodRate.text = popularModel.rate.toString() + ""
    }

    override fun getItemCount(): Int {
        return popularModel.size
    }

     class PopularFoodViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
       // var imageUrl: ImageView = itemView.findViewById(R.id.foodImage)
         var foodName: TextView = itemView.findViewById(R.id.foodName_tv)
         var foodDescription: TextView = itemView.findViewById(R.id.foodDesc_tv)
         var foodRate: TextView = itemView.findViewById(R.id.rate_tv)

     }

    fun setList(PopularModel: ArrayList<PopularModel>) {
        this.popularModel = PopularModel
    }
}